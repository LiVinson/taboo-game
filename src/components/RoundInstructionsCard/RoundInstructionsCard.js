import React from 'react'
import PropTypes from 'prop-types'
import { TabooCard, ButtonTabooCard } from 'components/shared/TabooCard'
import  InstructionsText  from 'components/shared/InstructionsText'
import  KeyWord  from 'components/shared/KeyWord'

//Depending on role of current users, returns a TabooCard with different set of instructions and button
//to start the round
const RoundInstructionsCard = ({ role, giver, watcher, startRound, error }) => {
	let instructions
	//Determine the instructions based on current player's team and role
	switch (role) {
		case 'giver':
			instructions = displayGiverInstructions(watcher, startRound, error)
			break
		case 'watcher':
			instructions = displayWatcherInstructions(giver)
			break
		case 'giverTeam':
			instructions = displayGiverTeamInstructions(giver, watcher)
			break
		case 'watcherTeam':
			instructions = displayWatcherTeamInstructions(giver, watcher)
			break
		default:
			instructions = ''
	}

	return instructions
}

RoundInstructionsCard.propTypes = {
	role: PropTypes.string.isRequired,
	giver: PropTypes.object.isRequired,
	watcher: PropTypes.object.isRequired,
	startRound: PropTypes.func.isRequired,
}

const displayGiverInstructions = (watcher, startRound, error) => {
	const buttonInfo = [
		{
			text: 'Start Round!',
			onClick: startRound,
		},
	]
	return (
		<ButtonTabooCard tabooWord="Get Ready!" buttons={buttonInfo} error={error}>
			<InstructionsText>
				It’s your turn to give clues to your team! {<KeyWord>{watcher.name}</KeyWord>} will be making sure you
				don’t say any taboo words.
			</InstructionsText>

			<InstructionsText>
				Select {<KeyWord>NEXT</KeyWord>} if your team guesses the word correctly or {<KeyWord>SKIP</KeyWord>} to go to the next card. Make sure to keep an
				eye on your time!
			</InstructionsText>
		</ButtonTabooCard>
	)
}

const displayWatcherInstructions = (giver) => {
	return (
		<TabooCard tabooWord="Get Ready!">
			<InstructionsText>
				It’s the other team’s turn to give clues! {<KeyWord>You</KeyWord>} are the watcher for this round! If{' '}
				{<KeyWord>{giver.name}</KeyWord>} says any words from the taboo list or gives any illegal clues, hit the{' '}
				{<KeyWord>buzzer!</KeyWord>}
			</InstructionsText>
		</TabooCard>
	)
}

const displayGiverTeamInstructions = (giver, watcher) => {
	return (
		<TabooCard tabooWord="Get Ready!">
			<InstructionsText>
				It's your turn to guess! {<KeyWord>{giver.name}</KeyWord>} will be giving the clues, and{' '}
				{<KeyWord>{watcher.name}</KeyWord>} will be checking for any illegal clues. Good luck!
			</InstructionsText>
		</TabooCard>
	)
}

const displayWatcherTeamInstructions = (giver, watcher) => {
	return (
		<TabooCard tabooWord="Just Relax!">
			<InstructionsText>
				It's the other team's turn to guess! {<KeyWord>{giver.name}</KeyWord>} will be giving the clues, and{' '}
				{<KeyWord>{watcher.name}</KeyWord>} will be checking for any illegal clues.
			</InstructionsText>
		</TabooCard>
	)
}
export default RoundInstructionsCard
