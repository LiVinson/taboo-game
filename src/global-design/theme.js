const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
