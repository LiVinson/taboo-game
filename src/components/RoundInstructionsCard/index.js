import RoundInstructionsCard from "./RoundInstructionsCard"

export default RoundInstructionsCard