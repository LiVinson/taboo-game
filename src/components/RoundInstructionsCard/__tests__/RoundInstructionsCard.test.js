import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import RoundInstructionsCard from "../RoundInstructionsCard"

test("RoundInstructionsCard renders correctly", ()=> {
	const props = {
		role: 'watcherTeam', 
        giver: { name: 'Sam', playerId: '12345', team: 'team 1' },
		watcher: { name: 'Jo', playerId: '23456', team: 'team 2' },
		startRound: jest.fn(),
	}

    const wrapper = renderer.create(
        <ThemeProvider theme={theme}>
            <RoundInstructionsCard {...props} />
        </ThemeProvider>
    ).toJSON()

    expect(wrapper).toMatchSnapshot()
})