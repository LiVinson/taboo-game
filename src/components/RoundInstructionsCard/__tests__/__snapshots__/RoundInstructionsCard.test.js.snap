// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundInstructionsCard renders correctly 1`] = `
<div
  className="sc-AxjAm hDlFTS"
>
  <div
    className="sc-AxiKw dGnLMo"
  >
    Just Relax!
  </div>
  <div
    className="sc-AxhCb JZaCe"
  >
    <p
      className="sc-fzokOt kxxFRs"
    >
      It's the other team's turn to guess! 
      <span
        className="sc-fzqBZW hgCjBU"
      >
        Sam
      </span>
       will be giving the clues, and
       
      <span
        className="sc-fzqBZW hgCjBU"
      >
        Jo
      </span>
       will be checking for any illegal clues.
    </p>
  </div>
</div>
`;
